File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        // n을 2진수로 변환
        String binaryN = Integer.toBinaryString(n);
        // n의 1의 개수를 저장
        int countOfN = countOfOne(binaryN);
        
        int num = n + 1;
        
        while(true){
            // num을 2진수로 변환 후 n과 1의 개수가 같은지 확인한다.
            if(countOfOne(Integer.toBinaryString(num))==countOfN){
                answer = num;
                break;
            }
            num++;
        }
        return answer;
    }
    // 2진수에서 1의 개수를 구하는 함수
    static int countOfOne(String binaryNum){
        int count=0;
        for(int i=0;i<binaryNum.length();i++){
            if(binaryNum.charAt(i)-'0'==1){
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    # n 다음으로 2진수 1갯수가 똑같은 수
    
    # 방법1 (직관적) 
    # n 1개수 저장
    # n 다음수들 1개수 세서 비교
    bin_cnt = bin(n)[2:].count('1')
    for i in range(n+1,10000001):
        if bin(i)[2:].count('1') == bin_cnt:
            return i
    
    # 방법2 
    # 0,1 규칙찾기
